javascripts/discourse/components/user-color-palette-selector.gjs

```diff
@@ -146,35 +146,32 @@ export default class UserColorPaletteSelector extends Component {
       `/color-scheme-stylesheet/${darkPaletteId}/${colorPalette.theme_id}.json`
     );
 
-    const replaceLinkTag = (oldTag, newHref, className, mode, paletteId) => {
+    Promise.all([
+      this.#preloadAndSwapCSS(lightPaletteInfo.new_href, "light-scheme"),
+      this.#preloadAndSwapCSS(darkPaletteInfo.new_href, "dark-scheme"),
+    ]).then(() => {
+      this.cssLoaded = true;
+    });
+  }
+
+  #preloadAndSwapCSS(newHref, existingLinkClass) {
```

Comment:
This probably eventually should move to core with the `InterfaceColorMode` stuff